main/php_syslog.c

```diff
@@ -105,6 +105,7 @@ PHPAPI void php_syslog(int priority, const char *format, ...) /* {{{ */
 #else
 PHPAPI void php_syslog(int priority, const char *format, ...) /* {{{ */
 {
+#ifndef __wasi__
```

Comment:
do you want to disable syslog completely or just some functions. Is this disabled just because of `php_openlog`? If so, you should just make that function noop.